### PR TITLE
Update to latest Ably SDK

### LINKED ISSUE #8: Upgrade ably-java to 0.8.8
This uses a very old version of ably-java (0.8.2) at the moment

### PR DESCRIPTION
**Pull request by funkyboy**
And Fix #8

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -27,8 +27,8 @@ repositories {
 }
 
 dependencies {
-    compile 'com.android.support:appcompat-v7:23.0.0'
-    compile 'com.android.support:percent:23.1.0'
-    compile 'io.ably:ably-java:0.8.2'
+    compile 'com.android.support:appcompat-v7:23.4.0'
+    compile 'com.android.support:percent:23.4.0'
+    compile 'io.ably:ably-java:1.0.2'
```
Comment (paddybyers) [inline app/build.gradle]: Not `ably-android` ?

**Review by paddybyers — CHANGES_REQUESTED** [review]
Just one comment; otherwise LGTM.

Thanks

**Comment by funkyboy** [comment]
You are right @paddybyers. I wonder why/how it worked anyway in the simulator:)

**Comment by paddybyers** [comment]
> I wonder why/how it worked anyway in the simulator:)

Because the ref you had will have correctly included the jar file from the java project, and 99% of the functionality is identical. The android variant gives us the ability to add platform-specific functionality (eg https://github.com/ably/ably-java/pull/292) but so far they're more or less identical code.

**Comment by funkyboy** [comment]
@paddybyers Oh I see. Anyway, fixed.

**Comment by paddybyers** [comment]
> so far they're more or less identical code

The other reason for separating them is so that we can run the tests on each of the separate platforms. Even though our code is the same, the implementations of the underlying platform libraries are different and so we need to test them independently.

**Pull request merged**